File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Embeddings/OpenAIEmbeddingsOptions.cs
using Microsoft.Teams.AI.Utilities;

namespace Microsoft.Teams.AI.AI.Embeddings
{
    /// <summary>
    /// Options for configuring an `OpenAIEmbeddings` to generate embeddings using an OpenAI hosted model.
    /// </summary>
    public class OpenAIEmbeddingsOptions : BaseOpenAIEmbeddingsOptions
    {
        /// <summary>
        /// API key to use when calling the OpenAI API.
        /// </summary>
        /// <remarks>
        /// A new API key can be created at https://platform.openai.com/account/api-keys.
        /// </remarks>
        public string ApiKey { get; set; }

        /// <summary>
        /// Model to use for embeddings.
        /// </summary>
        public string Model { get; set; }

        /// <summary>
        /// Optional. OpenAI organization.
        /// </summary>
        public string? Organization { get; set; }

        /// <summary>
        /// Initializes a new instance of the <see cref="OpenAIEmbeddingsOptions"/> class.
        /// </summary>
        /// <param name="apiKey">API key to use when calling the OpenAI API.</param>
        /// <param name="model">Model to use for embeddings.</param>
        public OpenAIEmbeddingsOptions(string apiKey, string model)
        {
            Verify.ParamNotNull(apiKey);
            Verify.ParamNotNull(model);

            ApiKey = apiKey;
            Model = model;
        }
    }
}

File: python/packages/ai/teams/ai/embeddings/openai_embeddings_options.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Dict, List, Optional


@dataclass
class OpenAIEmbeddingsOptions:
    """
    Options for configuring an `OpenAIEmbeddings` to generate embeddings.
    """

    api_key: str
    "API key to use. A new API key can be created at https://platform.openai.com/account/api-keys."

    model: str
    "Model to use."

    organization: Optional[str] = None
    "Optional. Organization to use."

    endpoint: Optional[str] = None
    "Optional. Endpoint to use."

    log_requests: Optional[bool] = False
    "Whether to log requests to the console, useful for debugging and defaults to `false`"

    retry_policy: Optional[List[int]] = None
    "Optional. Retry policy to use in seconds. The default retry policy is `[2, 5]`."

    request_config: Optional[Dict[str, str]] = None
    "Request options to use."
